integration-tests/test/basic-l1-l2-communication.spec.ts

```diff
@@ -28,48 +31,101 @@ describe('Basic L1<>L2 Communication', async () => {
       l2SimpleStorageJson.bytecode,
       env.l2Wallet
     )
+    Factory__L2Reverter = new ContractFactory(
+      l2ReverterJson.abi,
+      l2ReverterJson.bytecode,
+      env.l2Wallet
+    )
   })
 
   beforeEach(async () => {
     L1SimpleStorage = await Factory__L1SimpleStorage.deploy()
     await L1SimpleStorage.deployTransaction.wait()
     L2SimpleStorage = await Factory__L2SimpleStorage.deploy()
     await L2SimpleStorage.deployTransaction.wait()
+    L2Reverter = await Factory__L2Reverter.deploy()
+    await L2Reverter.deployTransaction.wait()
   })
 
-  it('should withdraw from L2 -> L1', async () => {
-    const value = `0x${'77'.repeat(32)}`
+  describe('L2 => L1', () => {
+    it('should be able to perform a withdrawal from L2 -> L1', async () => {
+      const value = `0x${'77'.repeat(32)}`
 
-    // Send L2 -> L1 message.
-    const transaction = await env.l2Messenger.sendMessage(
-      L1SimpleStorage.address,
-      L1SimpleStorage.interface.encodeFunctionData('setValue', [value]),
-      5000000
-    )
+      // Send L2 -> L1 message.
+      const transaction = await env.l2Messenger.sendMessage(
+        L1SimpleStorage.address,
+        L1SimpleStorage.interface.encodeFunctionData('setValue', [value]),
+        5000000
+      )
 
-    await env.waitForXDomainTransaction(transaction, Direction.L2ToL1)
+      await env.waitForXDomainTransaction(transaction, Direction.L2ToL1)
 
-    expect(await L1SimpleStorage.msgSender()).to.equal(env.l1Messenger.address)
-    expect(await L1SimpleStorage.xDomainSender()).to.equal(env.l2Wallet.address)
-    expect(await L1SimpleStorage.value()).to.equal(value)
-    expect((await L1SimpleStorage.totalCount()).toNumber()).to.equal(1)
+      expect(await L1SimpleStorage.msgSender()).to.equal(
+        env.l1Messenger.address
+      )
+      expect(await L1SimpleStorage.xDomainSender()).to.equal(
+        env.l2Wallet.address
+      )
+      expect(await L1SimpleStorage.value()).to.equal(value)
+      expect((await L1SimpleStorage.totalCount()).toNumber()).to.equal(1)
+    })
   })
 
-  it('should deposit from L1 -> L2', async () => {
-    const value = `0x${'42'.repeat(32)}`
+  describe('L1 => L2', () => {
```

Comment:
Adding tests for unsuccessful messages to dead addresses and 0x42 addrs would be nice to add here